File: DS/DS-Session2/CounsellingProgram/queue/Queue.java
package queue;
/**
 * It is an interface which has methods related to queue
 * 
 * @author Shubham Jain
 *
 */
public interface Queue<Type> {
    /**
     * For inserting an element in queue
     * 
     * @param data
     */
    void enQueue(Type data);

    /**
     * For removing an element from queue
     * 
     * @return
     */
    Type deQueue();

    /**
     * For checking whether queue is empty
     * 
     * @return
     */
    boolean isEmpty();

    /**
     * For checking whether queue is full
     * 
     * @return
     */
    boolean isFull();
}

File: UI/UI-Session8/queue.js
/**
 * Function to create class queue
 */
class Queue {
	constructor() {
		this.elements = [];
	}
}

/**
 * Function to check if the array is empty
 */
Queue.prototype.isEmpty = function() {
	if (this.elements.length == 0) {
        return true; 
	}
	return false;
}

/**
 * Function to add the element in the queue
 */
Queue.prototype.enqueue = function(data) {
	this.elements.push(data);
}

/**
 * Function to remove the element from the queue
 */
Queue.prototype.dequeue = function() {
	if (this.isEmpty()) {
        return "Queue Underflow"; 
	}
    return this.elements.shift(); 
}

/**
 * Function to display the element
 */
Queue.prototype.display = function() {
    for (var i = 0; i < this.elements.length; i++) { 
        console.log(this.elements[i]); 
    }
}